tests/test_auth_tokens.py

```diff
@@ -0,0 +1,94 @@
+# pylint: disable=no-self-use
+#       - pylint test classes must pass self, even if unused.
+# pylint: disable=logging-fstring-interpolation
+#       - honestly just annoying to use lazy(%) interpolation.
+# pylint: disable=redefined-outer-name
+#       - this is how we use fixtures internally so this throws false positives
+"""
+Used to test encoding/decoding JWT tokens
+and checking for their expiration
+"""
+from datetime import timedelta
+from typing import Dict, Any
+import time
+from fastapi.testclient import TestClient
+from app import app
+from models.auth import Token
+
+client = TestClient(app)
+
+
+def create_timedelta(minutes: int, sec: int, milli: int) -> timedelta:
+    """
+    Creates timedelta in order to encode token with different times.
+    """
+    delta = timedelta(minutes=minutes, seconds=sec, milliseconds=milli)
+    return delta
+
+
+class TestAuthUser:
+    def test_encode_payload_data_ok(self,
+                                    generate_random_str_data_dict: Dict[str,
+                                                                        Any]):
+        """
+        Creates a random dict and encodes it. It then
+        makes sure it returns an encoded string.
+        """
+        payload_dict = generate_random_str_data_dict
+        encoded_token = Token.get_enc_token_str_from_dict(payload_dict)
+        assert isinstance(encoded_token, str)
+
+    def test_decode_token(self, generate_random_str_data_dict: Dict[str, Any]):
+        """
+        Creates a random dict and encodes it. It then decodes it
+        and makes sure it returns a dict.
+        """
+        payload_dict = generate_random_str_data_dict
+        encoded_token_str = Token.get_enc_token_str_from_dict(
+            payload_dict)
+        decoded_dict = Token.get_dict_from_enc_token_str(
+            encoded_token_str)
+        assert isinstance(decoded_dict, dict)

```

Comment:
The method is good, but the check feels kind of small.

We should not only check that the decoded dict is a `dict` but also that it holds the same payload as the original dict.

That way we know if we lose any data in the encode/decode process.